RHEL6_7/system/requirements/check

```diff
@@ -104,26 +104,32 @@ get_free_size() {
 
 get_memory_size() {
   # MEMSIZE is in MB
-  MEMSIZE=$(dmidecode -t 17 \
-    | awk '( /Size/ && $2 ~ /^[0-9]+.B$/ ) {match($2, /^([0-9]+)(.B)$/, arr); $2=arr[1]; $3=arr[2]}
-           ( /Size/ && $3 ~ /^[kK]B$/ ) { unit=1.0/1024 }
-           ( /Size/ && $3 ~ /^MB$/ ) { unit=1 }
-           ( /Size/ && $3 ~ /^GB$/ ) { unit=2**10 }
-           ( /Size/ && $3 ~ /^TB$/ ) { unit=2**20 }
-           ( /Size/ && $2 ~ /^[0-9]+$/ ) { x+=int($2*unit) }
-           END{ print x}')
-  [ -n "$MEMSIZE" ] && return 0
-
-  #maybe virtual machine, so we can get only aroximation of real RAM size
-  log_warning "Can't get memory size with 'dmidecode'! Try 'dmesg' with 1% tolerance."
-  oo=$(dmesg | grep -ioE "System RAM:\s[0-9]+MB")
-  [ $? -eq 0 ] && {
-    MEMSIZE=$( echo "$oo" | cut -d " " -f 3 | cut -d " " -f 3 | sed s/..$// )
-    return 1
+  which dmidecode >/dev/null 2>&1 && {
+    # some archs doesn't contain dmidecode at all, so use it only when it
+    # exists
+    MEMSIZE=$(dmidecode -t 17 \
+      | awk '( /Size/ && $2 ~ /^[0-9]+.B$/ ) {match($2, /^([0-9]+)(.B)$/, arr); $2=arr[1]; $3=arr[2]}
+             ( /Size/ && $3 ~ /^[kK]B$/ ) { unit=1.0/1024 }
+             ( /Size/ && $3 ~ /^MB$/ ) { unit=1 }
+             ( /Size/ && $3 ~ /^GB$/ ) { unit=2**10 }
+             ( /Size/ && $3 ~ /^TB$/ ) { unit=2**20 }
+             ( /Size/ && $2 ~ /^[0-9]+$/ ) { x+=int($2*unit) }
+             END{ print x}')
+    [ -n "$MEMSIZE" ] && return 0
+  }
+
+  # this will not give us exact result, but still ok (~ 1% tolerance usually)
+
+  which dmesg >/dev/null 2>&1 && {
+    oo=$(dmesg | grep -ioE "System RAM:\s[0-9]+MB")
```

Comment:
On my system (RHEL 6.9, dmesg is installed) this variable is empty (regex does not match). In which case does it match. Also, try to use the variables with descriptive names.

Reply:
Yeah, that's because the lines are already out of the output the dmesg is providing (it!s still rolling during the run of the system). So you should be able to see that output after some time the system is running but you can loose it and that's why the "third" method to get the data is used (but the last one doesn't provide info about the real installed RAM so well...).

I am keeping the original names of variables as the module shouhld be rewritten. Until that time, I am keeping what is there and fixing only what needs to be really fixed by the PR.

Reply:
Output from the RHEL6 system.
```
# dmesg | grep -ioE "System RAM:\s[0-9]+MB"
System RAM: 2047MB
```